pkg/util/kube/priority.go

```diff
@@ -0,0 +1,383 @@
+// © Broadcom. All Rights Reserved.
+// The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+// SPDX-License-Identifier: Apache-2.0
+
+package kube
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	apitypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/priorityqueue"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	pkgnil "github.com/vmware-tanzu/vm-operator/pkg/util/nil"
+)
+
+// EventType represents a Kubernetes event.
+type EventType uint8
+
+const (
+	// EventCreate is the create event.
+	EventCreate EventType = iota
+	// EventUpdate is the update event.
+	EventUpdate
+	// EventDelete is the delete event.
+	EventDelete
+	// EventGeneric is the generic event.
+	EventGeneric
+)
+
+func (e EventType) String() string {
+	switch e {
+	case EventCreate:
+		return "create"
+	case EventUpdate:
+		return "update"
+	case EventDelete:
+		return "delete"
+	case EventGeneric:
+		return "generic"
+	}
+	return "unknown"
+}
+
+const (
+	// PriorityCreate is the priority for a create event.
+	PriorityCreate = -1 - iota
+	// PriorityUpdate is the priority for an update event.
+	PriorityUpdate
+	// PriorityDelete is the priority for a delete event.
+	PriorityDelete
+	// PriorityGeneric is the priority for a generic event.
+	PriorityGeneric
+)
+
+// GetPriorityFn is a function that may be used with a mapper that allows the
+// specification of a custom priority.
+type GetPriorityFn[T client.Object] func(
+	ctx context.Context,
+	eventType EventType,
+	newObj, oldObj T,
+	defaultPriority int) int
+
+var _ handler.EventHandler = &EnqueueRequestForObject{}
+
+// EnqueueRequestForObject enqueues a Request containing the Name and Namespace
+// of the object that is the source of the Event, e.g. the created / deleted /
+// updated objects Name and Namespace).
+type EnqueueRequestForObject = TypedEnqueueRequestForObject[client.Object]
+
+// TypedEnqueueRequestForObject enqueues a Request containing the Name and
+// Namespace of the object that is the source of the Event, e.g. the created /
+// deleted / updated objects Name and Namespace).
+type TypedEnqueueRequestForObject[T client.Object] struct {
+	Logger logr.Logger
+
+	// GetPriority returns the priority to use when submitting the object to the
+	// queue.
+	//
+	// Any value >0 will be submitted to the priority queue.
```

Comment:
I don't think we're checking the return value for this here

Reply:
Let me update the comment. All are submitted to the priority queue, just with different priorities.